### PR TITLE
[Suggest] enable explicit error for deepspeed not installed

### PR DESCRIPTION
**Pull request by Li-Qingyun**
If the user has not installed deepspeed and use deepspeed strategy, an implicit error of DeepSpeedStrategy not in REGISTRY will be raised.
Because of [here](https://github.com/open-mmlab/mmengine/blob/b2295a258c4af3cdeb1a55fa5e49cddbd1c6c0bf/mmengine/_strategy/__init__.py#L10C13-L10C13)

### CODE REVIEW

**Comment by zhouzaida** [comment]
Hi @Li-Qingyun , thanks for your contribution. We can consider registering the `DeepSpeedStrategy` into Strategy regardless of whether DeepSpeed is installed or not. We can then check for the installation of DeepSpeed in the `__init__` method.

**Comment by Li-Qingyun** [comment]
> Hi @Li-Qingyun , thanks for your contribution. We can consider registering the `DeepSpeedStrategy` into Strategy regardless of whether DeepSpeed is installed or not. We can then check for the installation of DeepSpeed in the `__init__` method.
> 
> Refer to https://github.com/open-mmlab/mmengine/blob/main/mmengine/optim/optimizer/apex_optimizer_wrapper.py

OKKKK! I have modified it following your great suggestion.

File: mmengine/_strategy/deepspeed.py [hunk 1]
```diff
@@ -71,6 +75,10 @@ def __init__(
         # the following args are for BaseStrategy
         **kwargs,
     ):
+        assert deepspeed is not None, \
+            'DeepSpeed is not installed. Please check ' \
+            'https://github.com/microsoft/DeepSpeed#installation.'
+
```
Comment (zhouzaida) [inline mmengine/_strategy/deepspeed.py:81]:
```suggestion
        assert deepspeed is not None, \
            'DeepSpeed is not installed. Please check ' \
            'https://github.com/microsoft/DeepSpeed#installation.'

```

File: mmengine/_strategy/__init__.py [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-from mmengine.utils import digit_version, is_installed
+from mmengine.utils import digit_version
 from mmengine.utils.dl_utils import TORCH_VERSION
 from .base import BaseStrategy
+from .deepspeed import DeepSpeedStrategy
 from .distributed import DDPStrategy
 from .single_device import SingleDeviceStrategy
 
-__all__ = ['BaseStrategy', 'DDPStrategy', 'SingleDeviceStrategy']
+__all__ = ['BaseStrategy', 'DDPStrategy', 'SingleDeviceStrategy', 'DeepSpeedStrategy']
```
Comment (zhouzaida) [inline mmengine/_strategy/__init__.py:9]:
```suggestion
__all__ = [
    'BaseStrategy', 'DDPStrategy', 'SingleDeviceStrategy', 'DeepSpeedStrategy'
]
```

**Comment by zhouzaida** [comment]
Also need to update the login in the following locations:

https://github.com/open-mmlab/mmengine/blob/b2295a258c4af3cdeb1a55fa5e49cddbd1c6c0bf/mmengine/optim/optimizer/__init__.py#L20

https://github.com/open-mmlab/mmengine/blob/b2295a258c4af3cdeb1a55fa5e49cddbd1c6c0bf/mmengine/optim/__init__.py#L38

**Comment by Li-Qingyun** [comment]
> Also need to update the login in the following locations:

@zhouzaida  OKKKK! Completed~!

**Comment by Li-Qingyun** [comment]
> Hi, the lint failed. You can refer to https://github.com/open-mmlab/mmengine/blob/main/CONTRIBUTING.md#2-configure-pre-commit to fix the lint.

Done~

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**